docs(STONEINTG-1304): add README.md with details of the repo (#55)
konflux-ci/konflux-operator-tasks: README.md

Refer to the [Building Tekton tasks as bundles in Konflux](https://konflux-ci.dev/docs/end-to-end/building-tekton-tasks/) guide for details on how to onboard the new Tekton tasks to Konflux, enabling their integration into the Konflux build pipelines.

#### Versioning

We follow a **versioning strategy** to ensure updates don’t break existing workflows.

A **new version** of a task should be created **if**:

- The task’s **interface changes** (e.g., parameters, workspaces, or result names are modified).
- New functionality is introduced that **isn’t backward compatible**.
- A critical bug fix **requires an updated implementation**.

Each version should be **clearly labeled** to avoid breaking existing pipelines. A `MIGRATION.md` file needs to be supplied containing instructions for migrating to the new version of the task.

### Policies

The `policies` directory contains the [Conforma](https://conforma.dev/docs/user-guide/index.html) policies that need to be applied when building bundle images of individual tasks.

### Task generator
The `task-generator` directory contains a collection of tools used for generating various versions of tasks.

### Scripts

> The `hack` directory contains the various scripts that are used during the CI runs of the build/test/release workflow for the individual Tekton tasks.

Reviewer: ```suggestion
The `hack` directory contains scripts for building, testing, and releasing the tasks.

```
